elastic/kibana
# [RAM] Rule List Tags Filtering with Infinite Scroll

## Summary
Resolves: https://github.com/elastic/kibana/issues/150364
Fixes: https://github.com/elastic/sdh-kibana/issues/3806

Fixes a common complaint and bug where we limit the number of tags that are displayed in the tags dropdown to 50. We now paginate these results with a max tag of 10,000 (50 per page). It's not a true pagination because Elasticsearch doesn't support filtering and paginating on aggregation buckets (bucket selector doesn't work on terms). Since tags are not nested properties or references, they can only be reached through terms aggregation.

But at least we don't return 10000 tags from the API.

## How to test:
1. By default we show 50 tags per page, to make testing easier, you may go to `x-pack/plugins/alerting/server/rules_client/methods/get_tags.ts` and set `DEFAULT_TAGS_PER_PAGE` to 10 or something
2. Create some rules
3. Create some tags (spread amongst the rules would be even better)
4. Open the tag filter
5. Should be able to search, paginate, and filter rules by the tags

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/74562234/217985463-ee3ce86f-e614-4690-a48a-74a328ca9cff.gif)

### Checklist
